apiserver/facades/client/controller/controller.go

```diff
@@ -104,9 +121,25 @@ func (s *ControllerAPIv3) checkHasAdmin() error {
 	return nil
 }
 
+// ModelStatus is a legacy method call to ensure that we preserve
+// backward compatibility.
+// TODO (anastasiamac 2017-10-26) This should be made obsolete/removed.
+func (s *ControllerAPIv3) ModelStatus(req params.Entities) (params.ModelStatusResults, error) {
+	results, err := s.ModelStatusAPI.ModelStatus(req)
+	if err != nil {
+		return params.ModelStatusResults{}, err
+	}
+	for _, r := range results.Results {
+		if r.Error != nil {
+			return params.ModelStatusResults{}, errors.Trace(r.Error)
```

Comment:
Did the v3 one actually return an empty results, or did it return a list of the right length?
I wouldn't be surprised if this is correct, I just wanted to confirm.

Reply:
The old return actually varied... Good pickup - from within the loop, a list of right length was returned.